### PR TITLE
refactor: New build system

### PR DESCRIPTION
**Pull request by flokX**
What does this PR accomplish?
- Move to Webpack with all-in-one build setup and development server
- Implement better theme compiling
- Clean up files and remove redundancies
- Update dependencies
- Use bootstrap-icons instead of Fontawesome

Open tasks:
- [x] Add GitHub actions build tasks (use [peaceiris/actions-gh-pages](https://github.com/peaceiris/actions-gh-pages)?)
- [x] Scrollspy navigation doesn't change focus when scrolling (To be fixed later when overhauling the page code)
- [x] Fix table appearance
- [x] Fix sections full-width white background
- [x] Adjust dynamic.js according to review result of #24

The benefits of this new build process are easier maintenance with automatic dependency updates when using Dependabot and automatic deployment.
Do you have interest in a Webpack and GitHub actions based build process? Please leave feedback.

### CODE REVIEW

**Comment by CaselIT** [comment]
Thanks for the help, but its best for first open an issue and/or discuss such large changes on [gitter](https://gitter.im/falconry/dev)

**Comment by flokX** [comment]
Hi, I thought it would be good to have a functional prototype to view and test. Therefore I drafted this PR so we can discuss about the changes.
The diff view looks quite messy because of the deleted and moved files due to consolidation work. For a good overview of the actual code changes for the new build system, see 1fce2f1907c4d2c566883da413dfa3ad584dbc84.

**Comment by CaselIT** [comment]
This is very appreciated, my remark was just to avoid possibly duplicating the work, example with https://github.com/falconry/falconry.github.io/pull/28

Maybe we could consider using bootstrap 5, even if currently in beta, since it removes the requirement for jquery (that I believe is currently used only by bootstrap).

Also I don't know if @kgriffs has some further ideas in mind, see also https://github.com/falconry/falcon/issues/1028

Feel free to drop a line on gitter

**Comment by vytas7** [comment]
AFAICT the proposed changeset looks sane an in-line with the modern FE practices based on Webpack, but I have to abstain on this. I'm simply too n00b at this stuff.

**Comment by flokX** [comment]
I have done everything related to the rewritten bundling process.
Please review especially the deploy process with the branch setup and the configuration keys as described in [github pages action](https://github.com/marketplace/actions/github-pages-action). After the other PRs are merged it would be nice to get some help to resolve the merge conflicts.

**Comment by flokX** [comment]
I hope you all had a good start to the new year. Is there any progress on the review of this PR?

File: package.json [hunk 1]
```diff
@@ -0,0 +1,30 @@
+{
+  "name": "falconry.github.io",
+  "version": "1.0.0",
+  "private": true,
+  "scripts": {
+    "serve": "webpack serve --port 8000 --config webpack-dev.config.js",
+    "build": "webpack --config webpack-prod.config.js"
+  },
+  "dependencies": {
+    "bootstrap": "^4.5.3",
```
Comment (CaselIT) [inline package.json:10]: I know bootstrap 5 is still in beta, but usually the bootstrap pre-release are still very stable.
Moving to v5 would also let us remove jquery

**Review by CaselIT — COMMENTED** [review]
I apparently have already remarked on bootstrap. I think it looks

File: package.json (hunk 1 above)
Comment (flokX) [reply package.json:10]: Technically your right. This PR already includes a lot of changes and I don't have that many resources right now, so I would propsose merging it and upgrade Bootstrap later.

**Review by CaselIT — APPROVED** [review]

**Review by kgriffs — APPROVED** [review]
Thanks! This was a lot of work, and very much appreciated!

**Pull request merged**